pkg/fetcher/fetcher.go

```diff
@@ -662,3 +688,77 @@ func (fetcher *Fetcher) SpecializePod(ctx context.Context, fetchReq FunctionFetc
 
 	return errors.Wrapf(err, "error specializing function pod after %v times", maxRetries)
 }
+
+// WebsocketHandler is used to generate websocket events in Kubernetes
+func (fetcher *Fetcher) WebsocketHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		http.Error(w, "only GET is supported on this endpoint", http.StatusMethodNotAllowed)
+		return
+	}
+	rec, err := eventRecorder(fetcher.kubeClient)
+	if err != nil {
+		klog.Errorf("Error creating recorder %s", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+	pods, err := fetcher.kubeClient.CoreV1().Pods(fetcher.Info.Namespace).List(metav1.ListOptions{
+		FieldSelector: "metadata.name=" + fetcher.Info.Name,
+	})
+	if err != nil {
+		fetcher.logger.Error("Failed to get the pod", zap.Error(err))
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+	for _, pod := range pods.Items {
```

Comment:
Ideally we should be getting only pod right? Can we not simply retrieve the first one?

Reply:
Yeah, length of `pod.Items` will always be 1 since we've used a field selector to obtain it. I don't think there's other way here